Sources/FoundationEssentials/String/String+Encoding+Names.swift

```diff
@@ -0,0 +1,265 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2025 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+
+// MARK: - Private extensions for parsing encoding names
+
+private extension Unicode.Scalar {
+    /// Returns the Boolean value that indicates whether or not `self` is "ASCII whitespace".
+    ///
+    /// Reference: https://infra.spec.whatwg.org/#ascii-whitespace
+    var _isASCIIWhitespace: Bool {
+        switch self.value {
+        case 0x09, 0x0A, 0x0C, 0x0D, 0x20: true
+        default: false
+        }
+    }
+}
+
+private extension String {
+    var _trimmed: Substring.UnicodeScalarView {
+        let scalars = self.unicodeScalars
+        let isNonWhitespace: (Unicode.Scalar) -> Bool = { !$0._isASCIIWhitespace }
+        guard let firstIndexOfNonWhitespace = scalars.firstIndex(where: isNonWhitespace),
+              let lastIndexOfNonWhitespace = scalars.lastIndex(where: isNonWhitespace) else {
+            return Substring.UnicodeScalarView()
+        }
+        return scalars[firstIndexOfNonWhitespace...lastIndexOfNonWhitespace]
+    }
+}
+
+/// A type that holds a `Unicode.Scalar` where its value is compared case-insensitively with others'
+/// _if the value is within ASCII range_.
+private struct ASCIICaseInsensitiveUnicodeScalar: Equatable,
+                                                  ExpressibleByUnicodeScalarLiteral {
+    typealias UnicodeScalarLiteralType = Unicode.Scalar.UnicodeScalarLiteralType
+
+    let scalar: Unicode.Scalar
```

Comment:
Can we store and operate every string processing logic on `UInt8` instead of `Unicode.Scalar` (and `UTF8View` instead of `UnicodeScalarView` since we expect it to be an ascii character?